### PR TITLE
fixed commerce demo error

### PR DESCRIPTION
**Pull request by pkghang**

### CODE REVIEW

File: web/modules/custom/simplytest_tugboat/src/PreviewConfigGenerator.php [hunk 1]
```diff
@@ -132,7 +132,7 @@ public function oneClickDemo(string $demo_id, array $parameters): array {
       ['echo "SIMPLYEST_STAGE_PATCHING"'],
       $one_click_demo->getPatchingCommands($parameters),
       [
-        'cd stm && composer update --no-ansi',
+        'cd stm && composer config allow-plugins true && composer update --no-ansi',
```
Comment (mglaman) [inline web/modules/custom/simplytest_tugboat/src/PreviewConfigGenerator.php:135]: We don't need to do this here, it's a bug due to the legacy Drupal Commerce project template.

File: web/modules/custom/simplytest_ocd/tests/src/Unit/CommerceConfigTest.php [hunk 2]
```diff
@@ -23,13 +23,14 @@ protected function getExpectedConfig(): array {
             'a2enmod headers rewrite',
             'rm -rf "${DOCROOT}"',
             'echo "SIMPLYEST_STAGE_DOWNLOAD"',
-            'cd "${TUGBOAT_ROOT}" && composer create-project drupalcommerce/demo-project stm --stability dev --no-interaction',
+            'cd "${TUGBOAT_ROOT}" && composer create-project drupalcommerce/demo-project stm --no-install --stability dev --no-interaction',
+            'composer config allow-plugins true && composer install',
             'ln -snf "${TUGBOAT_ROOT}/stm/web" "${DOCROOT}"',
             'echo "SIMPLYEST_STAGE_PATCHING"',
-            'cd stm && composer update --no-ansi',
+            'cd stm && composer config allow-plugins true && composer update --no-ansi',
```
Comment (mglaman) [inline web/modules/custom/simplytest_ocd/tests/src/Unit/CommerceConfigTest.php:30]: Setting `allow-plugins` shouldn't be needed here

**Review by mglaman — CHANGES_REQUESTED** [review]
A few requests and test failures

**Comment by mglaman** [comment]
Build error when manually testing

```
64f1c36e722387719c2cead8# /bin/sh -c cd "${TUGBOAT_ROOT}" && composer create-project drupalcommerce/demo-project stm --no-install --stability dev --no-interaction
Creating a "drupalcommerce/demo-project" project at "./stm"
Info from https://repo.packagist.org: #StandWithUkraine
Installing drupalcommerce/demo-project (dev-master edde71020649ee05caac4b6a08b7f5d0ecc6025f)
- Downloading drupalcommerce/demo-project (dev-master edde710)
- Installing drupalcommerce/demo-project (dev-master edde710): Extracting archive
Created project in /var/lib/tugboat/stm
64f1c36e722387719c2cead8# /bin/sh -c composer config allow-plugins true && composer install
In ConfigCommand.php line 217:
File "./composer.json" cannot be found in the current directory  
                                                                   

config [-g|--global] [-e|--editor] [-a|--auth] [--unset] [-l|--list] [-f|--file FILE] [--absolute] [-j|--json] [-m|--merge] [--append] [--source] [--] [<setting-key> [<setting-value>...]]
Command Failed (Tugboat Error 1064): Exit code: 1; Command: composer config allow-plugins true && composer install
```

**Review by mglaman — CHANGES_REQUESTED** [review]
Test failure

```
-            'image' => 'tugboatqa/php:7.3-apache'
+            'image' => 'tugboatqa/php:8.1-apache'
```

1) Drupal\Tests\simplytest_ocd\Unit\UmamiConfigTest::testConfigData

**Comment by pkghang** [comment]
@mglaman 
- "commerce-kickstart-project" project/plugin will integrate with new feature request ticket.
- PHPUnit test issue fixed
- Also updated the php version for umami test

File: web/modules/custom/simplytest_ocd/src/Plugin/OneClickDemo/Commerce.php [hunk 3]
```diff
@@ -28,7 +28,8 @@ public function getSetupCommands(array $parameters): array {
   public function getDownloadCommands($parameters): array {
     $commands = [
       // @todo the base preview doesn't have the `commerce` dir?
-      'cd "${TUGBOAT_ROOT}" && composer create-project drupalcommerce/demo-project stm --stability dev --no-interaction',
+      'cd "${TUGBOAT_ROOT}" && composer create-project drupalcommerce/demo-project stm --no-install --stability dev --no-interaction',
+      'composer config --global allow-plugins true && composer install',
```
Comment (mglaman) [inline web/modules/custom/simplytest_ocd/src/Plugin/OneClickDemo/Commerce.php:32]:
```
64f1e683961ebaf9ec9870c4# /bin/sh -c composer config --global allow-plugins true && composer install
Composer could not find a composer.json file in /var/lib/tugboat
To initialize a project, please create a composer.json file. See https://getcomposer.org/basic-usage
```

The previous command uses `'cd "${TUGBOAT_ROOT}" && ` this does as well

**Review by mglaman — CHANGES_REQUESTED** [review]

File: web/modules/custom/simplytest_ocd/tests/src/Unit/CommerceConfigTest.php [hunk 4]
```diff
@@ -23,13 +23,13 @@ protected function getExpectedConfig(): array {
             'a2enmod headers rewrite',
             'rm -rf "${DOCROOT}"',
             'echo "SIMPLYEST_STAGE_DOWNLOAD"',
-            'cd "${TUGBOAT_ROOT}" && composer create-project drupalcommerce/demo-project stm --stability dev --no-interaction',
+            'cd "${TUGBOAT_ROOT}" && composer create-project drupalcommerce/demo-project stm --no-install --stability dev --no-interaction',
+            'cd "${TUGBOAT_ROOT}/stm" composer config --global allow-plugins true && composer install',
             'ln -snf "${TUGBOAT_ROOT}/stm/web" "${DOCROOT}"',
             'echo "SIMPLYEST_STAGE_PATCHING"',
-            'cd stm && composer update --no-ansi',
```
Comment (mglaman) [inline web/modules/custom/simplytest_ocd/tests/src/Unit/CommerceConfigTest.php:29]: This is still happening, and should. That's why the test fails. I think.

**Review by mglaman — APPROVED** [review]
tests need to pass.

**Review by mglaman — CHANGES_REQUESTED** [review]
tests need to be fixed

**Pull request merged**